WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductSelectorViewModelTests.swift

```diff
@@ -345,7 +345,28 @@ final class ProductSelectorViewModelTests: XCTestCase {
         // Then
         let productRow = viewModel.productRows.first(where: { $0.productOrVariationID == product.productID })
         XCTAssertNotNil(productRow)
-        XCTAssertEqual(productRow?.selectedState, .selected)
+        if !ServiceLocator.featureFlagService.isFeatureFlagEnabled(.productMultiSelectionM1) {
```

Comment:
💡 For tests, it's recommended to use `MockFeatureFlagService` and inject it into the view models. That way, you can set the value for any flag and the tests will not fail if the actual flag is updated. You will not have to put the assertions in a conditional check either.

However, in this case we are testing `ProductSelectorViewModel`, which has been updated with the new flag, so you should inject the value into the initializer to test this case.

Reply:
> MockFeatureFlagService

Thanks for pointing that out, I didn't know about `MockFeatureFlagService` 🙇 

Injected the flag for all tests into the initializer on 82dfbb86c6d893a8526ce449998d7b438dc5c5c4 , rather than relying on the feature flag check.